core/roots.yml

```diff
@@ -8,21 +8,63 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
+# ------------------------------------------------------------------------------
 # This file defines all the different disk locations that this config uses.
+#
 # Root definitions are referenced from the toolkit configuration's templates.yml
 # file and folder creation schema.
+#
+# At setup time, each defined root is looked up against the local storages
+# defined in shotgun. For each root defined in this file, a local storage needs
+# to exist in Shotgun. The `shotgun_storage_id` key in each root definition
+# below should correspond to a local storage in Shotgun. This allows the roots
+# defined here to be called anything, making it possible to reuse the
+# configuration without changing references to the storage root, only the
+# associated SG storage id. If the `shotgun_storage_key` is not defined, then
+# the root name must match the name of the storage in SG (legacy fallback).
+#
+# NOTE: Local storages ids can only be retrieved via the API currently:
+#
+#    # get all local storages defined in SG
+#    shotgun.find("LocalStorage", [], ["code"])
+#
+# One of the roots defined in this file should have a `default` key with a value
+# of `true` to identify the default storage root to use when none is specified
+# for a path template. If no roots are identified a `default`, then one root
+# must be named `primary` which will indicate the default root.
+#
+# During setup, the paths defined for each local storage are transferred across
+# to this file.
+#
+# Note: In toolkit core versions prior to 0.18.121, configurations using a
+# single root required this root to be named 'primary'. In later versions of
+# core, this requirement has been lifted and the root can be named anything.
+# ------------------------------------------------------------------------------
 
-# At setup time, each defined root is looked up against the local storages defined
-# in shotgun. For each root defined in this file, a local storage with the same name
-# needs to exist in Shotgun. During setup, the paths defined for each local storage
-# are transferred across to this file.
+primary:
 
-# Note: In toolkit core versions prior to 0.18.121, configurations using a single
-# root required this root to be named 'primary'. In later versions of core, this
-# requirement has been lifted and the root can be named anything.
+    description: A top-level root folder for production data. Project folders
+        will be created beneath this location.
 
-primary:
-    description: "A top-level root folder for production data. Project folders will be created beneath this location."
+    # these paths will be populated by the associated local storage in Shotgun
+    # at setup time.
     mac_path:
     linux_path:
     windows_path:
+
+    # indicates that this is the default storage root
+    default: true
```

Comment:
as per @jfboismenu 's comment - will this be backwards comp with older versions of core that won't support it? Will it just be ignored? Worth testing to make sure (or bumping required core).

Reply:
If we are advertising these new values here, I think we should bump the core version.